presto-main/src/main/java/io/prestosql/server/DynamicFilterResource.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server;
+
+import io.airlift.log.Logger;
+import io.prestosql.operator.DynamicFilterSummary;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import java.util.Optional;
+import java.util.concurrent.Future;
+
+import static io.airlift.concurrent.MoreFutures.tryGetFutureValue;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+
+@Path("/v1/dynamic-filter/{queryId}/{source}")
+public class DynamicFilterResource
+{
+    private final DynamicFilterService service;
+
+    @Inject
+    public DynamicFilterResource(DynamicFilterService service)
+    {
+        this.service = service;
+    }
+
+    private static final Logger log = Logger.get(DynamicFilterResource.class);
+
+    @PUT
+    @Path("/{stageId}/{taskId}/{driverId}/{expectedCount}")
+    @Consumes(APPLICATION_JSON)
+    @Produces(TEXT_PLAIN)
+    public Response storeOrMergeSummary(
+            @PathParam("queryId") String queryId,
+            @PathParam("source") String source,
+            @PathParam("stageId") int stageId,
+            @PathParam("taskId") int taskId,
+            @PathParam("driverId") int driverId,
+            @PathParam("expectedCount") int expectedCount,
+            DynamicFilterSummary dynamicFilterSummary)
+    {
+        log.debug(String.format("Storing dynamic summary for queryId: %s, source %s, stageId: %d, taskId: %d, driverId: %d", queryId, source, stageId, taskId, driverId));
+        service.storeOrMergeSummary(queryId, source, stageId, taskId, driverId, dynamicFilterSummary, expectedCount);
```

Comment:
Should we validate the parameters before calling `service.storeOrMergeSummary(...)`?
For example, we can check that `0 <= driverId < expectedCount ` and that there is a running query `queryId`.